File: e037.js
function isprime(n) {
    if (n === 2 || n === 3) return true;
    if (n < 2 || n % 2 === 0) return false;
    if (n < 9) return true;
    if (n % 3 === 0) return false;

    let f = 5;
    while (f <= Math.floor(Math.sqrt(n))) {
        if (n % f === 0) return false;
        if (n % (f + 2) === 0) return false;
        f += 6;
    }
    return true;
}

let [n, count, s] = [11, 0, 0];
while (count < 11){
    if (isprime(n)){
        const a = n.toString();
        let k = 1;
        for (let i = 1; i<= a.length; i++){
            const nleft = Number(a.slice(i));
            const nright = Number(a.slice(0,i));
            if (isprime(nleft) && isprime(nright)) k += 1;
            else break;
        }
        if (k === a.length){
            s += n;
            count += 1;
            console.log(n)
        }
    }
    n += 2;
}
console.log(s);

File: e037.py
"""
Problem 37 - Truncatable primes
"""
from math import sqrt


def isprime(n):
    """Primality testing"""
    if n < 0: return print('input is negative')
    if n == 2 or n == 3: return True
    if n < 2 or n % 2 == 0: return False
    if n < 9: return True
    if n % 3 == 0: return False
    for f in range(5, int(sqrt(n)) + 1, 6):
        if n % f == 0 or n % (f + 2) == 0: return False
    return True


(n, count, s) = (11, 0, 0)
while count < 11:
    if isprime(n):
        a = str(n)
        k = 1
        for i in range(1, len(a)):
            nleft = int(a[i:])
            nright = int(a[:i])
            if isprime(nleft) and isprime(nright):
                k += 1
            else:
                break
        if k == len(a):
            s += n
            count += 1
            print(n, count)
    n += 2
print(s)
